### DISCUSSION TITLE
How to set domain for Strapi Admin Panel path on Heroku?

### DISCUSSION THREAD

**Discussion by faizananwerali** [strapi/strapi#7686] [Legacy Topics]
Recently I deployed my Strapi app on Heroku. If you look into routes https://example.herokuapp.com/ is for API and https://example.herokuapp.com/admin is for Admin Panel. Now I have my domain [mydomainexample.com][1] at Cloudflare which I can add in setting page. How can I use that [mydomainexample.com][1] domain in place of https://example.herokuapp.com/admin path? So whenever anyone visit [mydomainexample.com][1] it open admin panel and not show [herokuapp.com][2] domain. Maybe using nginx anyone?

[https://stackoverflow.com/q/63698871/3774282](https://stackoverflow.com/q/63698871/3774282)

  [1]: https://mydomainexample.com
  [2]: https://herokuapp.com

**Comment by derrickmehaffy** [comment]
Hello!

Please see the server.js configuration: https://strapi.io/documentation/v3.x/concepts/configurations.html#server

You will want to use the `url` key (not to be confused with the `admin.url` key). You can either set it statically or use the `env()` to load an environment variable.

**Reply by faizananwerali** [reply]
It didn't work.

_config/env/production/server.js_

`module.exports = ({ env }) => ({
    url: env('URL', '..'),
    admin: {
        url: env('ADMIN_URL', '/admin')
    }
});`

_Heroku Settings > Config Vars_
URL=https://api.example.com
ADMIN_URL=https//admin.example.com

URL did work but ADMIN_URL didn't work. visiting https//admin.example.com was showing nothing. I had to delete ADMIN_URL env to make it work as https://api.example.com/admin for admin panel which I don't want

**Reply by derrickmehaffy** [reply]
The ADMIN_URL will not be possible to use a sub-domain on Heroku as you need to have more control of how they handle the proxying (which you can't on Heroku) the only control you will have here is changing for example `/admin` to `/dashboard`.

If you need more control you will need to use a proxy software like Nginx hosted somewhere else that will proxy to your Heroku app.

**Comment by faizananwerali** [comment]
Will docker help in heroku? I know that in we can use nginx in docker.

**Reply by derrickmehaffy** [reply]
I am not sure here, as I'm not familiar with running Nginx on Heroku. We do have guides that explain the Nginx config for different types of environments here: https://strapi.io/documentation/v3.x/getting-started/deployment.html#optional-software-guides

But where you setup Nginx and how you apply those configs isn't something we will be able to directly assist with, it goes beyond the scope of Strapi itself and more on your platform.

**Comment by faizananwerali** [comment]
I read the documentation. It doesn't mention handing admin route to
subdomain using nginx or any other server. Even if I manage to do that all
strapi routes of admin panel will be affected. I think no one yet thought
of that.

**Reply by derrickmehaffy** [reply]
We do not have one for split sub-domains no but generally because it's not a very common setup. It would be fairly easy to do as you would just take the Sub-Folder-Split and tweak the config a bit for sub-domains instead.
